include/usb/usb_device.h

```diff
@@ -58,10 +58,12 @@ extern "C" {
 	static __in_section(usb, descriptor_##p, 2) __used __aligned(1)
 #define USBD_USER_DESCR_DEFINE(p) \
 	static __in_section(usb, descriptor_##p, 3) __used __aligned(1)
-#define USBD_STRING_DESCR_DEFINE(p) \
+#define USBD_INITIAL_STRING_DESCR_DEFINE(p) \
 	static __in_section(usb, descriptor_##p, 4) __used __aligned(1)
-#define USBD_TERM_DESCR_DEFINE(p) \
+#define USBD_STRING_DESCR_DEFINE(p) \
 	static __in_section(usb, descriptor_##p, 5) __used __aligned(1)
+#define USBD_TERM_DESCR_DEFINE(p) \
+	static __in_section(usb, descriptor_##p, 6) __used __aligned(1)
```

Comment:
Better the other way around with minimal changes, just:
```diff
diff --git a/include/usb/usb_device.h b/include/usb/usb_device.h
index 8884c8afbbe..83abab6755f 100644
--- a/include/usb/usb_device.h
+++ b/include/usb/usb_device.h
@@ -60,8 +60,10 @@ extern "C" {
        static __in_section(usb, descriptor_##p, 3) __used __aligned(1)
 #define USBD_STRING_DESCR_DEFINE(p) \
        static __in_section(usb, descriptor_##p, 4) __used __aligned(1)
-#define USBD_TERM_DESCR_DEFINE(p) \
+#define USBD_STRING_DESCR_USER_DEFINE(p) \
        static __in_section(usb, descriptor_##p, 5) __used __aligned(1)
+#define USBD_TERM_DESCR_DEFINE(p) \
+       static __in_section(usb, descriptor_##p, 6) __used __aligned(1)
 
 /*
  * This macro should be used to place the struct usb_cfg_data
```

Please revise you commit message, among others: "Without this fix, the order between individual string descriptors is undefined." is not correct. The order is well defined, you just need to understand how it works.

Reply:
What would be a better wording? The order might not be "undefined" per se, but from what I can tell there is currently no explicit constraint to ensure that the "initial" string descriptors are really placed first.

Reply:
Something like: "Add a new macro to define string descriptors in the user code more easily with the respect for the correct order."

Reply:
@abxy could you please follow-up to @jfischer-no's comment above?

Reply:
Sorry. The project where I needed this fix kind of fell to the wayside. I have changed the implementation like @jfischer-no suggested, and updated the commit message & PR description.

Reply:
Thank you @abxy